modules/ROOT/pages/field-level-capabilities.adoc

```diff
@@ -0,0 +1,256 @@
+component. If the value comes from a part of the parameter, the expression
+follows the selector to the desired field.
+
+The following is an example where a parameter named `body` has a field with
+values that depend on the acting parameter `workspace`:
+
+----
+public class ChatOperations {
+ public void sendChatMessage(@Connection ChatConnection chatConnection,
+       @FieldValues(value = ChannelsValueProvider.class, targetSelectors = "channelId",
+     bindings = {@Binding(actingParameter = "workspace",  extractionExpression = "body.routingInfo.workspace")}))
+       @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do message sending
+  }
+}
+----
+
+This acting parameter is taken from the same `body` parameter.
+
+The following shows the value provider used in this example:
+
+----
+public class ChannelsValueProvider implements ValueProvider {
+
+ @Connection
+ private ChatConnection chatConnection;
+
+ @Parameter
+ private String workspace;
+
+ @Override
+ public Set<Value> resolve() {
+   return ValueBuilder.getValuesFor(chatConnection.getChannels(workspace));
+ }
+
+ @Override
+ public String getId() {
+   return "Channels value provider";
+ }
+}
+----
+
+The `workspace` acting parameter defined in the value provider is bound to the
+extraction expression that takes the value of the field from the context
+of the component.
+
+== Sample Data Example
+
+The following is an example of a `@SampleData` annotation that binds an acting
+parameter to the value of a field of a parameter in the component:
+
+----
+public class ChatOperations {
+@SampleData(value = ChatMessageSampleDatatProvider.class, bindings = {@Binding(actingParameter = "workspace",  extractionExpression = "body.routingInfo.workspace")}))
+ public Result<InputStream, ChatAttributes> retrieveChatMessage(@Connection ChatConnection chatConnection,
+         @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do chat message retrieval
+  }
+}
+----
+
+In the *ChatMessageSampleDatatProvider*, there is a parameter named `workspace`
+that provides the sample data. The value of this acting parameter
+is taken from the extraction expression defined in the binding.
+
+== Extraction Expression Restrictions
+
+The extraction expression used for the extraction of the value of the acting
+parameter does not aggregate any data from the context. It must be limited to
+access only the value of a field in the parameter.
+
+The first part of this extraction expression must also correspond to the
+name of a parameter from the component.
+
+== Java Type Limitation
+
+When executing the extraction expression to give values to the acting
+parameters of a value provider or sample data provider, the output media type
+is always `application/java`. This means that the provider must not expect
+other media types, such as `application/json` or `application/xml`.
+
+== Parameters Static Metadata
```

Comment:
This seems like a topic that should appear much earlier in the set? Up around Line 144?